corehq/apps/app_manager/templatetags/xforms_extras.py

```diff
@@ -17,29 +17,35 @@ def translate(t, lang, langs=None):
 
 
 @register.filter
-def trans(name, langs=None, include_lang=True, use_delim=True):
+def trans(name, langs=None, include_lang=True, use_delim=True, prefix=False):
     langs = langs or ["default"]
     if include_lang:
         if use_delim:
-            suffix = lambda lang: ' [%s]' % lang
+            tag = lambda lang: ' [%s]' % lang
         else:
-            suffix = lambda lang: '''
+            tag = lambda lang: '''
                 <span class="btn btn-xs btn-info btn-langcode-preprocessed">%(lang)s</span>
             ''' % {"lang": html.escape(lang)}
     else:
-        suffix = lambda lang: ""
+        tag = lambda lang: ""
     for lang in langs:
         if lang in name and name[lang]:
-            return name[lang] + ("" if langs and lang == langs[0] else suffix(lang))
+            affix = ("" if langs and lang == langs[0] else tag(lang))
+            return affix + name[lang] if prefix else name[lang] + affix
         # ok, nothing yet... just return anything in name
     for lang, n in sorted(name.items()):
-        return n + suffix(lang)
+        affix = tag(lang)
+        return affix + n if prefix else n + affix
     return ""
 
 
 @register.filter
 def html_trans(name, langs=["default"]):
-    return mark_safe(html.strip_tags(trans(name, langs, use_delim=False)) or EMPTY_LABEL)
+    return mark_safe(trans(name, langs, use_delim=False) or EMPTY_LABEL)
```

Comment:
Removing `strip_tags` opens us up to javascript injection. Could fall back to doing text prefixes temporarily (`[en] Survey`) rather than deal with finding an alternative approach right now.